### DISCUSSION TITLE
Why can't I extend the CSF Args Type?

### DISCUSSION THREAD

**Discussion by motionrus** [storybookjs/storybook#20956] [General]
I have three **themes** that I transmit want to set globally in file `.storybook/preview.cjs`
```ts
export const parameters = {
  actions: { argTypesRegex: "^on[A-Z].*" },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
};

export const argTypes = {
  theme: { control: "select", options: ["labeler", "skill", "studio"] },
};
export const args = { theme: "labeler" };
```
In my story I use the decorator with data-attribute to define the theme
```ts
export default {
  title: "Example/Button",
  component: Button,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: {control: "color"},
  },
  decorators: [
    (Story, {args: {theme}}) => <div data-theme={theme}>{Story()}</div>,
  ],
} as ComponentMeta<typeof Button>;
```
I get an error when I try to create a story with the theme argument
```ts
export const Large = Template.bind({});

Large.args = {
  size: "large",
  label: "Button",
  // TS2322: Type '{ size: "large"; label: string; theme: string; }' is not assignable to type 'Partial<ButtonProps>'. 
  // Object literal may only specify known properties, and 'theme' does not exist in type 'Partial<ButtonProps>'
  theme: "studio",
};
```

**Comment by riledigital** [comment]
I don't have a good solution but I am having the same issue attempting to use custom decorator-only args with Storybook v7 + CSF3. I was able to sidestep this by using a custom `render` property instead, but using `render` to  feels counter to the docs. In fact, there is no top-level page for the `render` key, so I'm not sure what differentiates `render` from `decorators` as decorators seem to offer a superset of functionality. 

```jsx
render: (args) => 
<>
  <ContextProvider myProp={args.myCustomArg}>
    <ComponentName />
  </ContextProvider>
</>
```
🔗 [Decorators](https://storybook.js.org/docs/react/writing-stories/decorators#wrap-stories-with-extra-markup)
🔗 [Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf#custom-render-functions)
